extras/python/fogbench/__main__.py

```diff
@@ -174,12 +174,14 @@ def read_out_file(_file=None, _keep=False, _iterations=1, _interval=0, send_to='
         msg_transferred_itr = 0  # Messages transferred in every iteration
         byte_transferred_itr = 0  # Bytes transferred in every iteration
 
-        # Start sending the messages to server
-        _start_time.append(datetime.now())  # Start time of every iteration
-
         for idx, r in enumerate(readings_list):
             readings_list[idx]["key"] = str(uuid.uuid4())
+            msg_transferred_itr += 1
+            byte_transferred_itr += sys.getsizeof(r)
 
+        # Start sending the messages to server
+        _start_time.append(datetime.now())  # Start time of every iteration
+        for r in readings_list:
```

Comment:
This is a no: we need 2 loops.

Reply:
please elaborate!

Reply:
As for the comment in my previews review. 

CURRENT:
for r in readings_list:
            is_sent = False
            if send_to == 'coap':
                is_sent = loop.run_until_complete(send_to_coap(r))
            elif send_to == 'http':
                is_sent = loop.run_until_complete(send_to_http(r))

            if not is_sent:
                break

CORRECT:
if send_to == 'coap':
    _start_time.append(datetime.now())
   for r in readings_list:
       loop.run_until_complete(send_to_coap(r))
elif send_to == 'http':
    _start_time.append(datetime.now())
    for r in readings_list:
       loop.run_until_complete(send_to_http(r))
_end_time.append(datetime.now())

I am not sure about the returned value for run_until_complete - we did not check it in the previous version.